octodns/provider/constellix.py

```diff
@@ -19,59 +18,106 @@
 
 
 class ConstellixClientException(Exception):
-    pass
+    pass  # gas
 
 
 class ConstellixClientBadRequest(ConstellixClientException):
-
     def __init__(self, resp):
-        errors = resp.json()['errors']
         super(ConstellixClientBadRequest, self).__init__(
-            '\n  - {}'.format('\n  - '.join(errors)))
+            '\n  - {}'.format('\n  - '.join(resp.json()['errors']))
+        )
 
 
 class ConstellixClientUnauthorized(ConstellixClientException):
-
     def __init__(self):
         super(ConstellixClientUnauthorized, self).__init__('Unauthorized')
 
 
 class ConstellixClientNotFound(ConstellixClientException):
-
     def __init__(self):
         super(ConstellixClientNotFound, self).__init__('Not Found')
 
 
 class ConstellixClient(object):
-    BASE = 'https://api.dns.constellix.com/v1/domains'
+    """
+    REST client for interacting with Constellix DNS API service
+    """
+    BASE = 'https://api.dns.constellix.com/v1'
 
     def __init__(self, api_key, secret_key, ratelimit_delay=0.0):
+        """
+        Constructor
+        :param api_key: string - The user's API key
+        :param secret_key: string - The user's secret key that is associated
+         with the API key
+        :param ratelimit_delay: float - The rate-limit delay in milliseconds
+        (i.e. 1 / max_rate or 1 / max_frequency)
+        """
         self.api_key = api_key
         self.secret_key = secret_key
         self.ratelimit_delay = ratelimit_delay
         self._sess = Session()
-        self._sess.headers.update({'x-cnsdns-apiKey': self.api_key})
         self._domains = None
 
-    def _current_time(self):
-        return str(int(time.time() * 1000))
-
-    def _hmac_hash(self, now):
-        return hmac.new(self.secret_key.encode('utf-8'), now.encode('utf-8'),
-                        digestmod=hashlib.sha1).digest()
-
-    def _request(self, method, path, params=None, data=None):
-        now = self._current_time()
-        hmac_hash = self._hmac_hash(now)
+    @staticmethod
```

Comment:
Not sure all these `@staticmethod` are necessary, most of the methods are only relevant within an instance of the ConstellixProvider class.